### ISSUE TITLE
Are vault.enpassdbsync supported?

### ISSUE THREAD

**Issue by PhantomPhoton** [costastf/enpassreadercli#1]
Howdy! I have enpass version 6.9.3 and I created a vault via the webdav backend. It created a vault.enpassdbsync file which does not work with this utility. I've verified the number of rounds match the value embedded in the database file and that the key is the same key I give enpass to unlock.

Is this expected?

**Comment by costastf** [comment]
Hi @PhantomPhoton , I have no clue what that file is and I cannot find any concrete definition of it or is usage (https://support.enpass.io/docs/security-whitepaper-enpass/search.html?q=enpassdbsync&check_keywords=yes&area=default for example). Does the library work on your main db file as expected?

**Comment by costastf** [comment]
It seems to be a custom format with some metadata as headers and binary content. Probably would need to look into this quite a bit.

**Comment by PhantomPhoton** [comment]
So I'm only using the webdav sync method and all my vaults have been created as this file type :( I've tried using the next cloud backend (which only allows one vault) and it created the same file.

Thank you for looking into it and if support happens, I'll be happy to use your utility. If not, I completely understand

**Comment by costastf** [comment]
Can you please try something for me as I am a bit sick at the moment and can't look into it myself? From what I see the vault.enpassdbsync file seems to be a binary file with a header of some json which in the normal case would be the contents of vault.json . Can you please try to make a local copy of your sync file, edit it with any text editor and remove the json payload and create a vault.json file with the contents and the binary contents of the other file leave as is and rename to vault.enpassdb and then try the CLI on the resulting file. I am pretty sure that it will work and will of course validate it myself when I am better but if you can manage to do it and it works we know upfront that it is trivial to fix.

**Comment by PhantomPhoton** [comment]
So I tried removing the json blob and in-between the json blob there's a lot of nulls (`0x00`). I tried removing those as well, but neither file opened successfully.

I'll try to generate a normal vault and see if I can match up the file formats at all.

**Comment by PhantomPhoton** [comment]
I'm on a Mac, and the vaults do get synced and stored into `~/Library/Containers/in.sinew.Enpass-Desktop/Data/Documents/Vaults` in three files `vault.enpassdb` `vault.enpassdn` and `vault.json`

`vault.enpassdn` is a empty file

`vault.json` does match the header on the `vault.enpassdbsync` file

The nulls do appear to be a padding. They pad the json to exactly 1024 bytes.

The data blob doesn't match exactly though. Diffing their hex dump doesn't show a clear delta, it's a bunch of small data chunks mismatched thought the two files.

I'll keep at it and see if I can gleam something else out, but this is the limits of my knowledge.

Hope you feel better soon!

**Comment by costastf** [comment]
Thanks for the wishes and the research! I am assuming that the tool works as expected on the vault.enpassdb file you have locally? If that is the case, at least you are unblocked for now and we have more time to look into the issue.

**Comment by PhantomPhoton** [comment]
Yes, the tool works great on the cached vault.enpassdb file. Thank you so much.

**Comment by costastf** [comment]
Thank you for reporting! Actually that is not the cache file, it is your actual vault, the remote one is the sync file. This tool is targeting that vault. I will look into supporting the sync file if it's easy to do.

**Comment by PhantomPhoton** [comment]
A update! I asked a buddy of mine for some help and I messed up the other tries somehow.

`dd if=vault.enpassdbsync of=vault.enpassdb bs=1024 skip=1` and then loading the created vault.enpassdb does appear to work.

**Comment by costastf** [comment]
Awesome! I was hoping/expecting this to be honest. As soon as I find some time I will extend the tool to work on the sync file then! Thanks for this!
